edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.CloudProxy.Test/ConnectivityAwareClientTest.cs

```diff
@@ -37,6 +37,14 @@ void ConnectionStatusChangedHandler(ConnectionStatus status, ConnectionStatusCha
             deviceConnectivityManager.InvokeDeviceConnected();
             deviceConnectivityManager.InvokeDeviceDisconnected();
 
+            // Assert
+            Assert.Equal(0, connectionStatusChangedHandlerCount);
+
+            // Act
+            await connectivityAwareClient.OpenAsync();
+            deviceConnectivityManager.InvokeDeviceConnected();
+            deviceConnectivityManager.InvokeDeviceDisconnected();
+
             // Assert
             Assert.Equal(2, connectionStatusChangedHandlerCount);
 
```

Comment:
If the test is to check the handler is removed when connection closed then we should only assert after CloseAsync. I would say that for other tests, it should only fail if the action that is tested fails